### DISCUSSION TITLE
issue: setValue() set field that dose noit exist in the form.

### DISCUSSION THREAD

**Discussion by JolianGof** [react-hook-form/react-hook-form#11275] [Q&A]
### Version Number

7.45.4

### Codesandbox/Expo snack

https://stackblitz.com/edit/stackblitz-starters-sou4kd?file=src%2FApp.js

### Steps to reproduce

1. Go to 'App.js'
2. Click on 'setValue'
3. It going to show an text
4. See my issue

### Expected behaviour

I would like to know if my usage is correct. I am using the setValue() function as shown in the code below. It seems to be working even though the field doesn't exist in the form.

### What browsers are you seeing the problem on?

Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
While calling `setValue` on a non-existent field may update its value initially, [this is unsupported](https://react-hook-form.com/docs/useform/setvalue#:~:text=It%20will%20not%20create%20a%20new%20field%20when%20targeting%20a%20non%2Dexisting%20field.) and could lead to unexpected behavior.

> It will not create a new field when targeting a non-existing field.

To handle additional fields that are not "user-facing," use input [type="hidden"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/hidden).

**Comment by JolianGof** [comment]
In our project, we use it to add additional state that we want to use across multiple components, much like a store.

I want aware that in future versions, it is anticipated to undergo changes? @Moshyfawn

**Comment by Moshyfawn** [comment]
What kind of state are you storing in the non-existent fields? It seems like it would be more appropriate to store your external state in a React state / context.

I can't promise you this behaviour is not going to change since this is not a supported RHF feature.
